plugins/modules/docker_compose_v2.py

```diff
@@ -0,0 +1,924 @@
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+
+- name: Run with inline Compose file version 2
+  # https://docs.docker.com/compose/compose-file/compose-file-v2/
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - community.docker.docker_compose:
+        project_src: flask
+        state: absent
+
+    - community.docker.docker_compose:
+        project_name: flask
+        definition:
+          version: '2'
+          services:
+            db:
+              image: postgres
+            web:
+              build: "{{ playbook_dir }}/flask"
+              command: "python manage.py runserver 0.0.0.0:8000"
+              volumes:
+                - "{{ playbook_dir }}/flask:/code"
+              ports:
+                - "8000:8000"
+              depends_on:
+                - db
+      register: output
+
+    - ansible.builtin.debug:
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+
+- name: Run with inline Compose file version 1
+  # https://docs.docker.com/compose/compose-file/compose-file-v1/
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - community.docker.docker_compose:
+        project_src: flask
+        state: absent
+
+    - community.docker.docker_compose:
+        project_name: flask
+        definition:
+            db:
+              image: postgres
+            web:
+              build: "{{ playbook_dir }}/flask"
+              command: "python manage.py runserver 0.0.0.0:8000"
+              volumes:
+                - "{{ playbook_dir }}/flask:/code"
+              ports:
+                - "8000:8000"
+              links:
+                - db
+      register: output
+
+    - ansible.builtin.debug:
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+'''
+
+RETURN = '''
+stdout:
+  description:
+    - The stdout from docker-compose.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: str
+stderr:
+  description:
+    - The stderr from docker-compose.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: str
+containers:
+  description:
+    - A dictionary mapping the various status of containers during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) was not given the chance to run
+  type: complex
+  contains:
+    container_name:
+      description: Name of the container.
+      returned: always, unless when C(docker-compose) was not given the chance to run
+      type: list
+      elements: str
+      example: ["stopped", "removed"]
```

Comment:
If `containers` is a dictionary mapping strings (container names) to lists of strings (states of containers), then the above should be replaced by:
```suggestion
  sample:
    container_1:
      - stopped
      - removed
    container_2:
      - created
      - started
```

Reply:
You are currently documenting a structure like
```
containers:
  container_name:
    - stopped
    - removed
```
where both `containers` and `container_name` are fixed strings and `containers` only ever contains one element (namely `container_name`).

Reply:
Thanks a lot. I tried to make sense of this by reading Ansible's documentation, which is lacking on this matter IMO, then the source-code, which is very "bare-bones" regarding plugins documentation processing and doesn't explain much of what is _functionally_ expected.
From what you are saying, there is no way to document arbitrary keys for a dict with the `contains` structure ?

Reply:
> From what you are saying, there is no way to document arbitrary keys for a dict with the `contains` structure ?

Yes, that's not possible.

Reply:
Should be fixed. I retained the description.